### PR TITLE
creating Edit profile page and integrate it with the backend #23 #36

### PR DESCRIPTION
**Pull request by Abdulrahimch**
### What this PR does (required):
- Create edit profile page 
- make it responsive
- addressing comments provided by Sunday and Rajiv.
- Integrate Edit profile page. 

kindly have a look at the following video:
https://user-images.githubusercontent.com/44474691/137955040-cda138c9-6d24-47fa-bff9-18430e77cf94.mp4

dependencies:
1-"@date-io/date-fns": "1.3.13"
2-"@material-ui/pickers": "3.3.10"
3-"date-fns": "2.25.0

"
myProfile05
myProfile04
myProfile03
myProfile02
myprofile01

### CODE REVIEW

**Review by ezeilo-su — COMMENTED** [review]
Looks good! Hope to see the screenshot(s) of your results uploaded in your future PRs.
Happy coding!

File: client/src/pages/EditProfile/EditProfile.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    const inputs = { firstName, lastName, gender, birthDate, phoneNumber, address, description };
+    const id = auth.loggedInUser ? auth.loggedInUser.profile : '';
+    editProfile(inputs, id).then((data) => {
+      if (data.error) {
+        console.log(data.error);
+        updateSnackBarMessage(data.error);
+      } else if (data.success) {
+        updateSnackBarMessage('Your profile has been updated successfully');
+      } else {
+        updateSnackBarMessage('An unexpected error occurred. Please try again later');
```
Comment (rajivtitus) [inline client/src/pages/EditProfile/EditProfile.tsx:37]: Good usage of the snackbar 👍🏽

File: client/src/pages/EditProfile/useStyles.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  },
+  gridItem: {
+    marginTop: '1.5em',
+  },
+  typography: {
+    fontWeight: 700,
+    [theme.breakpoints.down('md')]: {
+      fontSize: 10,
+    },
+    textTransform: 'uppercase',
```
Comment (rajivtitus) [inline client/src/pages/EditProfile/useStyles.ts:25]: 👍🏽

File: client/src/pages/EditProfile/EditProfile.tsx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+import { editProfile } from '../../helpers/APICalls/profile';
+
+import { useSnackBar } from '../../context/useSnackbarContext';
+import { useAuth } from '../../context/useAuthContext';
+
+export default function EditProfile(): JSX.Element {
+  const classes = useStyles();
+  const { updateSnackBarMessage } = useSnackBar();
+
+  const [firstName, setFirstName] = useState<string>('');
```
Comment (rajivtitus) [inline client/src/pages/EditProfile/EditProfile.tsx:17]: Why don't we consider using 'Formik' for a better managed state and form validation? :)

**Comment by rajivtitus** [comment]
> > Good work with this! I've left some minor comments for you to address.
> > Question: How do we plan to create a profile for a user?
> 
> Well, We can use the same page after changing the title to be "My Profile" instead of "Edit Profile" and adding some validations to ensure that the user has entered all fields. we also modify the tab of "Edit Profile" to be "My Profile".
> 
> I have come up with this plan just not to repeat all fields in a seprate page. Does it make sense?

Yes, it does make sense. Since we had a discussion about this during our standup meeting today, I will approve this PR. Fetah and yourself can finalize the logic for profile creation :)

File: client/src/pages/EditProfile/EditProfile.tsx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+  const [lastName, setLastName] = useState<string>('');
+  const [gender, setGender] = useState<string>('male');
+  const [birthDate, setBirthDate] = useState<Date | null>(new Date('1988-05-15'));
+  const [phoneNumber, setPhoneNumber] = useState<string>('');
+  const [address, setAddress] = useState<string>('');
+  const [description, setDescription] = useState<string>('');
+
+  const auth = useAuth();
```
Comment (rajivtitus) [inline client/src/pages/EditProfile/EditProfile.tsx:25]:
```suggestion
  const { loggedInUser } = useAuth();
```

File: client/src/pages/EditProfile/EditProfile.tsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+  const onSave = async () => {
+    const inputs = { firstName, lastName, gender, birthDate, phoneNumber, address, description };
+    const id = auth.loggedInUser ? auth.loggedInUser.profile : '';
```
Comment (rajivtitus) [inline client/src/pages/EditProfile/EditProfile.tsx:29]:
```suggestion
    if (loggedInUser.profile) {
    editProfile(inputs, id).then((data)....
    ......
   }
```

**Pull request merged**